indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -1800,25 +1800,25 @@ public Response getRowStats(
   )
   {
     IndexTaskUtils.datasourceAuthorizationCheck(req, Action.READ, getDataSource(), authorizerMapper);
-    return Response.ok(doGetRowStatsAndUnparseableEvents(full, false).lhs).build();
+    return Response.ok(doGetRowStatsAndUnparseableEvents(full != null, false).lhs).build();
   }
 
   @VisibleForTesting
```

Comment:
I think we can remove this test hook and directly call the API methods. It would help cleaning up the code a bit and also provide more test coverage that CI is complaining about. It would need passing in an additional mocked request object from the callers.